v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubStreamLT.java

```diff
@@ -0,0 +1,155 @@
+
+  private static PubsubResourceManager pubsubResourceManager;
+  private static ArtifactClient gcsClient;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void test1Backlog10Gb() throws IOException {
+    testBacklog(Function.identity(), 30, 60);
+  }
+
+  public void testBacklog(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder,
+      long generatorTimeOut,
+      long pipelineTimeOut)
+      throws IOException {
+    // Creates a topic on Pub/Sub.
+    TopicName outputTopic = pubsubResourceManager.createTopic("topic");
+    // Creates a subscription on Pub/Sub.
+    SubscriptionName outputSubscription =
+        pubsubResourceManager.createSubscription(outputTopic, "output-subscription");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaTemplate(testName, "GAME_EVENT")
+            .setQPS("1000000")
+            .setSinkType("GCS")
+            .setOutputDirectory(getTestMethodDirPath())
+            .setNumShards("20")
+            .setNumWorkers("20")
+            .setMaxNumWorkers("50")
+            .build();
+
+    // Executes the data generator and return approximate number of messages
+    // Multiply the number by 0.99 to get 99% of total messages
+    long expectedMessages =
+        (long) (dataGenerator.execute(Duration.ofMinutes(generatorTimeOut)) * NINETY_NINE_PERCENT);
+
+    LaunchConfig options =
+        paramsAdder
+            .apply(
+                LaunchConfig.builder(testName, SPEC_PATH)
+                    .addParameter("outputTopic", outputTopic.toString())
+                    .addParameter("inputFilePattern", getTestMethodDirPath() + "/*"))
+            .build();
+
+    // Act
+    PipelineLauncher.LaunchInfo info = pipelineLauncher.launch(project, region, options);
+    // Check if the pipeline runs
+    assertThatPipeline(info).isRunning();
+    try {
+      // Get the result of running pipeline
+      PipelineOperator.Result result =
+          pipelineOperator.waitUntilDone(createConfig(info, Duration.ofMinutes(pipelineTimeOut)));
+      assertNotEquals(PipelineOperator.Result.LAUNCH_FAILED, result);
+      // Check if the jobs runs
+      assertEquals(
+          PipelineLauncher.JobState.RUNNING,
+          pipelineLauncher.getJobStatus(project, region, info.jobId()));
+      // Checks if the collection has the expected number of messages.
+      assertTrue(waitForNumMessages(info.jobId(), INPUT_PCOLLECTION, expectedMessages));
```

Comment:
Should we export metrics? (as it was implemented in other LTs)
Example:
`exportMetricsToBigQuery(info, getMetrics(info, INPUT_PCOLLECTION, OUTPUT_PCOLLECTION));`

Reply:
I don't know. 
@Lizzfox , could you please, help us.
Thank you.

Reply:
Yep, we should export metric results.